src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -98,8 +124,9 @@ public void storeMessage(Message message) {
         String text = (String) entity.getProperty("text");
         long timestamp = (long) entity.getProperty("timestamp");
         double sentimentScore = (double) entity.getProperty("sentimentScore");
+        String parentKey = (String) entity.getProperty("parentKey");
```

Comment:
Parent key is retrieved using: https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/Entity#getParent